www/include/monitoring/common-Func.php

```diff
@@ -79,208 +86,157 @@ function set_user_param($user_id, $pearDB, $key, $value)
     $_SESSION[$key] = $value;
 }
 
-function get_notified_infos_for_host($hostId)
+/**
+ * @param $hostId
+ * @param $dependencyInjector
+ * @return array
+ */
+function getNotifiedInfosForHost($hostId, $dependencyInjector)
 {
     global $pearDB;
-    $loop = array();
-    $stack = array($hostId);
-    $hosts = array();
+    $hostInstance = Host::getInstance($dependencyInjector);
     $results = array('contacts' => array(), 'contactGroups' => array());
-    $stopReading = array('contacts' => 0, 'contactGroups' => 0);
-
-    while (($hostId = array_shift($stack))) {
-        if (isset($loop[$hostId])) {
-            continue;
-        }
-        $loop[$hostId] = 1;
-
-        $DBRESULT = $pearDB->query("SELECT contact_additive_inheritance, cg_additive_inheritance
-                FROM host WHERE host_id = " . $hostId);
-        $contactAdd = $DBRESULT->fetchRow();
 
-        /*
-         * Manage contact inheritance
-         */
-        $contactGroups = getContactGroupsForHost($hostId);
-        $contacts = getContactsForHost($hostId);
-
-        if ($stopReading['contacts'] == 0) {
-            $results['contacts'] = $results['contacts'] + $contacts;
-        }
-        if ($stopReading['contactGroups'] == 0) {
-            $results['contactGroups'] = $results['contactGroups'] + $contactGroups;
-        }
-
-        if ($contactAdd['contact_additive_inheritance'] == 0 && count($contacts) > 0) {
-            $stopReading['contacts'] = 1;
-        }
-        if ($contactAdd['cg_additive_inheritance'] == 0 && count($contactGroups) > 0) {
-            $stopReading['contactGroups'] = 1;
-        }
+    $query = "SELECT * FROM host WHERE host_id = " . (int)$hostId;
```

Comment:
seems a bit overkill to get all information from host, instead of needed columns